File: packages/square-mobile-payments/android/src/main/java/io/capawesome/capacitorjs/plugins/squaremobilepayments/classes/options/StartPaymentOptions.java
package io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.CustomExceptions;

public class StartPaymentOptions {

    @NonNull
    private final PaymentParameters paymentParameters;

    @NonNull
    private final PromptParameters promptParameters;

    public StartPaymentOptions(@NonNull PluginCall call) throws Exception {
        JSObject paymentParametersObj = call.getObject("paymentParameters");
        if (paymentParametersObj == null) {
            throw CustomExceptions.PAYMENT_PARAMETERS_MISSING;
        }
        this.paymentParameters = new PaymentParameters(paymentParametersObj);

        JSObject promptParametersObj = call.getObject("promptParameters");
        if (promptParametersObj == null) {
            throw CustomExceptions.PROMPT_PARAMETERS_MISSING;
        }
        this.promptParameters = new PromptParameters(promptParametersObj);
    }

    @NonNull
    public PaymentParameters getPaymentParameters() {
        return paymentParameters;
    }

    @NonNull
    public PromptParameters getPromptParameters() {
        return promptParameters;
    }
}

File: packages/square-mobile-payments/ios/Plugin/Classes/Options/StartPaymentOptions.swift
import Foundation
import Capacitor

@objc public class StartPaymentOptions: NSObject {
    let paymentParameters: PaymentParameters
    let promptParameters: PromptParameters

    init(_ call: CAPPluginCall) throws {
        guard let paymentParametersObj = call.getObject("paymentParameters") else {
            throw CustomError.paymentParametersMissing
        }
        self.paymentParameters = try PaymentParameters(paymentParametersObj)

        guard let promptParametersObj = call.getObject("promptParameters") else {
            throw CustomError.promptParametersMissing
        }
        self.promptParameters = try PromptParameters(promptParametersObj)
    }
}
